<launch>
    # specify the needle tracking mode
    <!-- <arg name = "pub_mode" default = "0"/>

    <node name="davinci_needle_pose_publisher" pkg="cwru_davinci_grasp" type="davinci_needle_pose_publisher_main" args="$(arg pub_mode)" respawn="false" output="screen">
        <param name="trans_noise_mean"    type="double" value="0.002"/>
        <param name="trans_noise_stddev"  type="double" value="0.0001"/>
        <param name="orient_noise_mean"   type="double" value="0.02"/>
        <param name="orient_noise_stddev" type="double" value="0.001"/>
    </node> -->

    # specify which arm is going to perform needle grasp
    <arg name = "which_arm" default = "psm_one"/>

    # specify the name of the needle
    <arg name = "needle_name" default = "needle_r"/>

    # specify to pick or place needle
    <arg name = "movement" default = "pick"/>

    <node name="davinci_simple_needle_grasper_main" pkg="cwru_davinci_grasp" type="davinci_simple_needle_grasper_main" args="$(arg which_arm) $(arg needle_name) $(arg movement)" respawn="false" output="screen">
        <rosparam command="load" file="$(find cwru_davinci_grasp)/config/dvrk_psm_grasp_needle_data.yaml"/>
        <param name="/planning_plugin" value="ompl_interface/OMPLPlanner"/>
        <rosparam command="load" file="$(find cwru_davinci_dvrk_both_psms_moveit_config)/config/ompl_planning.yaml"/>
    </node>
</launch>
